cocotb/regression.py

```diff
@@ -180,34 +185,41 @@ def _discover_tests() -> Iterable[Test]:
                 _logger.info(traceback.format_exc())
                 raise
 
-            if test_str:
-
+            if tests:
                 # Specific functions specified, don't auto-discover
-                for test_name in test_str.rsplit(','):
-                    try:
-                        test = getattr(module, test_name)
-                    except AttributeError:
-                        _logger.error("Requested test %s wasn't found in module %s", test_name, module_name)
-                        err = AttributeError("Test %s doesn't exist in %s" % (test_name, module_name))
-                        raise err from None  # discard nested traceback
-
-                    if not isinstance(test, Test):
-                        _logger.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
-                                      test_name, module_name)
-                        raise ImportError("Failed to find requested test %s" % test_name)
-
-                    # If we request a test manually, it should be run even if skip=True is set.
-                    test.skip = False
-
-                    yield test
+                for test_name in tests:
+                    test = getattr(module, test_name, 0)
+                    if test:
+                        if not isinstance(test, Test):
+                            _logger.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
+                                          test_name, module_name)
+                            raise ImportError("Failed to find requested test %s" % test_name)
+
+                        # If we request a test manually, it should be run even if skip=True is set.
+                        test.skip = False
+
+                        yield test
+                    else:
+                        not_found_tests.append(test_name)
```

Comment:
paired with the suggestion above:
```suggestion
```

Reply:
Changed in 7b89f41f006ed5c7da2b03838072de5474024bbf.